keystore/src/entities/platform/generic/mls/e2ei_crl.rs

```diff
@@ -67,41 +67,26 @@ impl EntityBase for E2eiCrl {
     }
 
     async fn save(&self, conn: &mut Self::ConnectionType) -> crate::CryptoKeystoreResult<()> {
-        use rusqlite::OptionalExtension as _;
         use rusqlite::ToSql as _;
 
         Self::ConnectionType::check_buffer_size(self.content.len())?;
 
         let transaction = conn.transaction()?;
-        let existing_rowid = transaction
-            .query_row(
-                "SELECT rowid FROM e2ei_crls WHERE distribution_point = ?",
-                [self.distribution_point.as_str()],
-                |r| r.get::<_, i64>(0),
-            )
-            .optional()?;
 
         let zb = rusqlite::blob::ZeroBlob(self.content.len() as i32);
 
-        let row_id = if let Some(row_id) = existing_rowid {
-            let params: [rusqlite::types::ToSqlOutput; 3] =
-                [self.distribution_point.to_sql()?, zb.to_sql()?, row_id.to_sql()?];
+        // Use UPSERT (ON CONFLICT DO UPDATE)
+        let sql = "
+        INSERT INTO e2ei_crls (distribution_point, content) 
+        VALUES (?, ?) 
+        ON CONFLICT(distribution_point) DO UPDATE SET content = excluded.content
+        RETURNING rowid";
```

Comment:
[Disclaimer: I'm a db newbie.]

Is there a way to prepare the transaction beforehand so that the SQL is parsed only once and then we just execute the prepared transaction with parameters?

The way this is currently done (and also before) looks like the SQL is going to be parsed anytime `save` is called.
I'm not sure how performance critical this function is, but it may be a good idea use a prepared statement anyway.

Reply:
rusqlite have some API for caching statements but let's maybe look into that in separate PR?